documentation/private-link-service/private-link-redirect-map.md

```diff
@@ -0,0 +1,249 @@
+
+The only difference is that the IP address used in underlay routing and 4to6 action will be specified by a port-based service mapping – for a specified port range, different IP or destination port can be used for crafting the packet.
+
+![PL Redirect VM-to-PLS direction](images/private-link-redirect-vm-to-pls.svg)
+
+On the reverse direction, it reverses all the actions in the forwarding side, e.g., 6to4, as it shows below:
+
+![PL Redirect PLS-to-VM direction](./images/private-link-redirect-pls-to-vm.svg)
+
+### 3.2 Redirect map with PL NSG
+When PL NSG is enabled, the extra encap for tunneling the packet to NSG will still be added on top the original PL encap. And the return packet will be exactly the same as the regular case without PL NSG.
+
+### 3.3 Redirect map with fast path
+The Fast Path here is not ER Fast Path, but the fast path that required by PLS. Fast path ICMP flow redirection will still take effect when redirect map is used.
+
+- If PL NSG is not used, it changes the flow just like regular PL case.
+
+- If PL NSG is used, it only updates the PL encap, leaving the NSG encap unchanged.
+
+For more information on how Fast Path works, please refer to [Fast Path documentations](../load-bal-service/fast-path-icmp-flow-redirection.md).
+
+### 3.4 Non-required features 
+- RST on connection idle timeout.
+- Flow resimulation should not be triggered automatically on redirect map update.
+
+## 4. Resource modeling, requirement, and SLA
+### 4.1 Scaling requirement
+The scaling requirement for PL redirect map are listed as below. The metrics are based on a 200Gbps DPU:
+| Metric | Requirement |
+| ---    | --- |
+| # of ENIs per DPU | 32 |
+| # of VNET mapping per ENI  | 64K |
+| # of PPS | 64M |
+| # of redirect mapping per DPU | 2000 maps (Reusable among all ENIs) |
+| # of entries in each redirect mapping | 8K |
+| VNET mapping change rate (CRUD) | <TBD> |
+| Redirect map entry change rate (CRUD) | Monthly |
+| # of fast path packets | Same as CPS. 3M per card. |
+
+### 4.2 Reliability requirements
+The flows created when redirect map will be replicated among multiple DPUs, following the SmartSwitch HA design.
+
+For more information, please refer to [SmartSwitch HA design doc](https://github.com/sonic-net/SONiC/blob/master/doc/smart-switch/high-availability/smart-switch-ha-hld.md).
+
+## 5. DASH object model design
+Now we have all action defined, we can use these actions to define our overall DASH pipeline.
+
+### 5.1 From On-Premise to PLS direction
+Say, we have a VM in on-premises network with IP 10.0.0.1, trying to reach the Private Endpoint in their VNET with IP 10.0.1.1, and the ExpressRoute VNI is 1000000:
+
+#### 5.1.1 Private Link
+1. **VNI Lookup**: First, we will look up the VNI to determine the packet direction. In this case, we consider all the packets from on-premises network as outbound direction from the floating NIC perspective.
+```json
```

Comment:
it will be better to describe the things in the form of SAI API in DASH, and leaving the SONiC dash tables into the SONiC-DASH HLD as DB schema and etc.

Reply:
updated